addons/info/src/index.js

```diff
@@ -29,59 +34,62 @@ const defaultMarksyConf = {
-    if (typeof storyFn !== 'function') {
-      if (typeof info === 'function') {
+export function addInfo(storyFn, context, info, _options) {
+  if (typeof storyFn !== 'function') {
+    if (typeof info === 'function') {
         _options = storyFn; // eslint-disable-line
         storyFn = info; // eslint-disable-line
         info = ''; // eslint-disable-line
-      } else {
-        throw new Error('No story defining function has been specified');
-      }
+    } else {
+      throw new Error('No story defining function has been specified');
     }
+  }
 
-    const options = {
-      ...defaultOptions,
-      ..._options,
-    };
+  const options = {
+    ...defaultOptions,
+    ..._options,
+  };
 
-    // props.propTables can only be either an array of components or null
-    // propTables option is allowed to be set to 'false' (a boolean)
-    // if the option is false, replace it with null to avoid react warnings
-    if (!options.propTables) {
-      options.propTables = null;
-    }
+  // props.propTables can only be either an array of components or null
+  // propTables option is allowed to be set to 'false' (a boolean)
+  // if the option is false, replace it with null to avoid react warnings
+  if (!options.propTables) {
+    options.propTables = null;
+  }
 
-    const marksyConf = { ...defaultMarksyConf };
-    if (options && options.marksyConf) {
-      Object.assign(marksyConf, options.marksyConf);
-    }
+  const marksyConf = { ...defaultMarksyConf };
+  if (options && options.marksyConf) {
+    Object.assign(marksyConf, options.marksyConf);
+  }
+  const props = {
+    info,
+    context,
+    showInline: Boolean(options.inline),
+    showHeader: Boolean(options.header),
+    showSource: Boolean(options.source),
+    propTables: options.propTables,
+    propTablesExclude: options.propTablesExclude,
+    styles: typeof options.styles === 'function' ? options.styles : s => s,
+    marksyConf,
+    maxPropObjectKeys: options.maxPropObjectKeys,
+    maxPropArrayLength: options.maxPropArrayLength,
+    maxPropsIntoLine: options.maxPropsIntoLine,
+    maxPropStringLength: options.maxPropStringLength,
+  };
+  return (
+    <Story {...props}>
+      {storyFn(context)}
+    </Story>
+  );
+}
 
-    return this.add(storyName, context => {
-      const props = {
-        info,
-        context,
-        showInline: Boolean(options.inline),
-        showHeader: Boolean(options.header),
-        showSource: Boolean(options.source),
-        propTables: options.propTables,
-        propTablesExclude: options.propTablesExclude,
-        styles: typeof options.styles === 'function' ? options.styles : s => s,
-        marksyConf,
-        maxPropObjectKeys: options.maxPropObjectKeys,
-        maxPropArrayLength: options.maxPropArrayLength,
-        maxPropsIntoLine: options.maxPropsIntoLine,
-        maxPropStringLength: options.maxPropStringLength,
-      };
+export const withInfo = (info, _options) =>
```

Comment:
- [ ] This will need a followup in https://github.com/DefinitelyTyped/DefinitelyTyped/